sgkit/distance/metrics.py

```diff
@@ -172,3 +175,234 @@ def correlation_reduce(v: ArrayLike, out: ArrayLike) -> None:  # pragma: no cove
     if denom > 0:
         value = 1 - (num / denom)
     out[0] = value
+
+
+def call_metric_kernel(
+    f: ArrayLike, g: ArrayLike, metric: str, metric_kernel: Any
+) -> ArrayLike:  # pragma: no cover.
+    # Numba's 0.54.0 version is required, which is not released yet
+    # We install numba from numba conda channel: conda install -c numba/label/dev numba
+    # Relevant issue https://github.com/numba/numba/issues/6824
+    f = np.ascontiguousarray(f)
+    g = np.ascontiguousarray(g)
+
+    # move input data to the device
+    d_a = cuda.to_device(f)
+    d_b = cuda.to_device(g)
+    # create output data on the device
+    out = np.zeros((f.shape[0], g.shape[0], N_MAP_PARAM[metric]), dtype=f.dtype)
+    d_out = cuda.to_device(out)
+
+    # TODO: Consider using cupy for directly creating zeros on GPU
```

Comment:
Can you add an issue for this?

Reply:
I'll remove this Todo, this was a note just for myself to try cupy.